aten/src/ATen/native/cuda/MultiLabelMarginCriterion.cu

```diff
@@ -0,0 +1,446 @@
+            using accscalar_t = at::acc_type<scalar_t, true>;
+            multilabel_margin_loss_forward_kernel<scalar_t, accscalar_t>
+                <<<blocks, threads, 0, at::cuda::getCurrentCUDAStream()>>>(
+                    output.data_ptr<scalar_t>(),
+                    input_.data_ptr<scalar_t>(),
+                    target_.data_ptr<int64_t>(),
+                    is_target_.data_ptr<scalar_t>(),
+                    nframe,
+                    dim,
+                    false);
+            C10_CUDA_KERNEL_LAUNCH_CHECK();
+          });
+    }
+
+  } else {
+    TORCH_CHECK(
+        false,
+        "Expected 2D input with optional zero batch dim, or 1D input with non-zero dims, but got sizes: ",
+        input.sizes());
+  }
+}
+
+void multilabel_margin_loss_backward_cuda_out_template(
+    const Tensor& grad_output,
+    const Tensor& input,
+    const Tensor& target,
+    int64_t reduction,
+    const Tensor& is_target,
+    Tensor& grad_input) {
+  check_shape(input, target);
+  auto input_ = input.contiguous();
+  if (input_.numel() == 0) {
+    return;
+  }
+
+  grad_input.resize_as_(input_);
+  auto target_ = target.contiguous();
+  auto is_target_ = is_target.contiguous();
+  auto grad_output_ = grad_output.contiguous();
+
+  if (grad_input.dim() <= 1) {
+    int dim = grad_input.dim() == 0 ? 1 : grad_input.size(0);
+    int target_size = target_.dim() == 0 ? 1 : target_.size(0);
+    TORCH_CHECK(
+        (target_.numel() != 0) && (target_.dim() <= 1) && (target_size == dim),
+        "inconsistent target size");
+    TORCH_CHECK(
+        target_.sizes() == is_target_.sizes(), "inconsistent is_target size");
+    dim3 blocks(1);
+    dim3 threads(MULTILABELMARGIN_THREADS);
+
+    AT_DISPATCH_FLOATING_TYPES_AND2(
+        at::ScalarType::Half,
+        at::ScalarType::BFloat16,
+        input.scalar_type(),
+        "multilabel_margin_loss_backward_kernel",
+        [&] {
+          using accscalar_t = at::acc_type<scalar_t, true>;
+          multilabel_margin_loss_backward_kernel<scalar_t, accscalar_t>
+              <<<blocks, threads, 0, c10::cuda::getCurrentCUDAStream()>>>(
+                  grad_input.data_ptr<scalar_t>(),
+                  grad_output_.data_ptr<scalar_t>(),
+                  input_.data_ptr<scalar_t>(),
+                  target_.data_ptr<int64_t>(),
+                  is_target_.data_ptr<scalar_t>(),
+                  1,
+                  dim,
+                  reduction == at::Reduction::Mean,
+                  reduction != at::Reduction::None);
+          C10_CUDA_KERNEL_LAUNCH_CHECK();
+        });
+  } else if (grad_input.dim() == 2) {
+    int nframe = grad_input.size(0);
+    int dim = grad_input.size(1);
+    TORCH_CHECK(
+        (input_.size(1) != 0) && (target_.dim() == 2) &&
+            (target_.size(0) == nframe) && (target_.size(1) == dim),
+        "inconsistent target size");
+    TORCH_CHECK(
+        (is_target_.dim() == 2) && (is_target.size(0) == nframe) &&
```

Comment:
nit: this check is slightly cleaner to write as `target_.size() == is_target_.size()`